src/exe/Main.hs

```diff
@@ -198,53 +196,48 @@ loadImport decl = do
     $ loadModule srcSpan modName
 
 -- | Loads the interface of the module with the given name from the
---   configured import path.
+--   configured import path or base library.
 --
 --   The given source span is used in error messages if the module could
---   not be found.
+--   not be found. Modules from the base library always take precedence.
+--   Allowed file extensions for module interface files are @.json@ (for
+--   generated module interfaces) and @.toml@ (for manually maintained
+--   module interface files).
 loadModule :: SrcSpan -> IR.ModName -> Application ()
 loadModule srcSpan modName = do
+  baseLibDir <- inOpts optBaseLibDir
   importDirs <- inOpts optImportDirs
-  ifaceFile <- findIfaceFile importDirs
+  ifaceFile <- findIfaceFile (baseLibDir : importDirs)
   iface <- loadModuleInterface ifaceFile
   modifyEnv $ makeModuleAvailable iface
  where
   -- | The name of the module's interface file relative to the import
   --   directories.
   filename :: FilePath
-  filename = map (\c -> if c == '.' then '/' else c) modName <.> "json"
+  filename = map (\c -> if c == '.' then '/' else c) modName
+
+  -- | The allowed file extensions of module interface files.
+  extensions :: [String]
+  extensions = ["json", "toml"]
 
   -- | Looks for the module's interface file in the import directories.
   --
   --   Reports a fatal message if the file could not be found.
   findIfaceFile :: [FilePath] -> Application FilePath
   findIfaceFile []       = reportFatal
     $ Message srcSpan Error
-    $ "Could not find imported module " ++ showPretty modName
+    $ "Could not find imported module " ++ showPretty modName ++ "."
   findIfaceFile (d : ds) = do
-    let ifaceFile = d </> filename
-    exists <- liftIO $ doesFileExist ifaceFile
-    if exists then return ifaceFile else findIfaceFile ds
+    let ifaceFiles = [d </> filename <.> ext | ext <- extensions]
```

Comment:
I think I would be more readable to also move the `ds` argument into the list comprehension and use a `case` on the result of `findM`.